arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -0,0 +1,334 @@
+            break;
+          case INT_LONG_BACKED_DECIMAL:
+            ((IcebergArrowVectors.DecimalArrowVector) vec).setNullabilityHolder(nullabilityHolder);
+            vectorizedColumnIterator.nextBatchIntLongBackedDecimal(vec, typeWidth, nullabilityHolder);
+            break;
+          case VARBINARY:
+            ((IcebergArrowVectors.VarBinaryArrowVector) vec).setNullabilityHolder(nullabilityHolder);
+            vectorizedColumnIterator.nextBatchVarWidthType(vec, nullabilityHolder);
+            break;
+          case VARCHAR:
+            ((IcebergArrowVectors.VarcharArrowVector) vec).setNullabilityHolder(nullabilityHolder);
+            vectorizedColumnIterator.nextBatchVarWidthType(vec, nullabilityHolder);
+            break;
+          case FIXED_WIDTH_BINARY:
+            ((IcebergArrowVectors.VarBinaryArrowVector) vec).setNullabilityHolder(nullabilityHolder);
+            vectorizedColumnIterator.nextBatchFixedWidthBinary(vec, typeWidth, nullabilityHolder);
+            break;
+          case BOOLEAN:
+            vectorizedColumnIterator.nextBatchBoolean(vec, nullabilityHolder);
+            break;
+          case INT:
+            vectorizedColumnIterator.nextBatchIntegers(vec, typeWidth, nullabilityHolder);
+            break;
+          case LONG:
+            vectorizedColumnIterator.nextBatchLongs(vec, typeWidth, nullabilityHolder);
+            break;
+          case FLOAT:
+            vectorizedColumnIterator.nextBatchFloats(vec, typeWidth, nullabilityHolder);
+            break;
+          case DOUBLE:
+            vectorizedColumnIterator.nextBatchDoubles(vec, typeWidth, nullabilityHolder);
+            break;
+        }
+      }
+    }
+    Preconditions.checkState(vec.getValueCount() == numValsToRead,
+        "Number of values read, %s, does not equal expected, %s", vec.getValueCount(), numValsToRead);
+    return new VectorHolder(columnDescriptor, vec, allPagesDictEncoded, dictionary, nullabilityHolder);
+  }
+
+  private void allocateFieldVector() {
+    if (allPagesDictEncoded) {
+      Field field = new Field(
+          icebergField.name(),
+          new FieldType(icebergField.isOptional(), new ArrowType.Int(Integer.SIZE, true), null, null),
+          null);
+      this.vec = field.createVector(rootAlloc);
+      ((IntVector) vec).allocateNew(batchSize);
+      typeWidth = IntVector.TYPE_WIDTH;
+    } else {
+      PrimitiveType primitive = columnDescriptor.getPrimitiveType();
+      if (primitive.getOriginalType() != null) {
+        switch (columnDescriptor.getPrimitiveType().getOriginalType()) {
+          case ENUM:
+          case JSON:
+          case UTF8:
+          case BSON:
+            this.vec = new IcebergArrowVectors.VarcharArrowVector(icebergField.name(), rootAlloc);
+            //TODO: Possibly use the uncompressed page size info to set the initial capacity
+            vec.setInitialCapacity(batchSize * 10);
```

Comment:
Should 10 be a `private static final` variable instead of hard-coded here? Looks like this is supposed to be the default average value size for varchars.